ground/src/main/java/com/google/android/ground/persistence/sync/LocalMutationSyncWorker.kt

```diff
@@ -59,61 +58,33 @@ constructor(
 
   private suspend fun doWorkInternal(): Result =
     try {
-      val mutations = getPendingOrEligibleFailedMutations()
-      Timber.d("Syncing ${mutations.size} changes for LOI $locationOfInterestId")
-      val result = processMutations(mutations)
+      val user = userRepository.getAuthenticatedUser()
+      val mutations =
+        mutationRepository
+          .getMutationsEligibleForRetry(locationOfInterestId, MAX_RETRY_COUNT)
+          .filter {
+            if (it.userId != user.id) Timber.w("ignoring mutation $it belonging to another user")
+            it.userId == user.id
+          }
+      Timber.d(
+        "Syncing ${mutations.size} changes authored by user ${user.id} on LOI $locationOfInterestId"
+      )
+      val result = processMutations(mutations, user)
       mediaUploadWorkManager.enqueueSyncWorker(locationOfInterestId)
       if (result) success() else retry()
     } catch (t: Throwable) {
       Timber.e(t, "Failed to sync changes for LOI $locationOfInterestId")
       retry()
     }
 
-  /**
-   * Attempts to fetch all mutations from the [MutationRepository] that are in `PENDING` state or in
-   * `FAILED` state but eligible for retry.
-   */
-  private suspend fun getPendingOrEligibleFailedMutations(): List<Mutation> {
-    val pendingMutations =
-      mutationRepository.getMutations(locationOfInterestId, MutationEntitySyncStatus.PENDING)
-    val failedMutationsEligibleForRetry =
-      mutationRepository
-        .getMutations(locationOfInterestId, MutationEntitySyncStatus.FAILED)
-        .filter { it.retryCount < MAX_RETRY_COUNT }
-    return pendingMutations + failedMutationsEligibleForRetry
-  }
-
-  /**
-   * Groups mutations by user id, loads each user, applies mutations, and removes processed
-   * mutations.
-   *
-   * @return `true` if all mutations are applied successfully, else `false`
-   */
-  private suspend fun processMutations(allMutations: List<Mutation>): Boolean {
-    val mutationsByUserId = allMutations.groupBy { it.userId }
-    val userIds = mutationsByUserId.keys
-    var noErrors = true
-    for (userId in userIds) {
-      val mutations = mutationsByUserId[userId]
-      val user = getUser(userId)
-      if (mutations == null || user == null) {
-        continue
-      }
-      val result = processMutations(mutations, user)
-      if (!result) {
-        noErrors = false
-      }
-    }
-    return noErrors
-  }
-
   /**
    * Applies mutations to remote data store. Once successful, removes them from the local db.
    *
    * @return `true` if the mutations were successfully synced with [RemoteDataStore].
    */
   private suspend fun processMutations(mutations: List<Mutation>, user: User): Boolean {
-    check(mutations.isNotEmpty()) { "List of mutations is empty" }
+    // TODO(#2235): Process and update each mutation individually.
```

Comment:
I think we realized yesterday we may need to process them together so we can build a batch. Ok to remove this comment.